### DISCUSSION TITLE
How to check if a response is redirect response?

### DISCUSSION THREAD

**Discussion by juniorforlife** [remix-run/remix#6074] [Q&A] [answered]
Hi @sergiodxa, in your article https://sergiodxa.com/articles/throwing-vs-returning-responses-in-remix,
you mentioned 

> Throwing a response can cause two effects, depending on whether the response is a redirect.
> 
>     1. If you throw a redirect response, Remix will follow the redirect as usual.
>     2. If you throw a non-redirect response, Remix will render the CatchBoundary component of the route (or nearest parent route) and pass the response data to the UI with the useCatch hook.

But how do we check if a response is a redirect response or not?

**Comment by sergiodxa** [comment, marked as answer]
If you have a Response object check if [status is a redirect one](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#redirection_messages), and if the headers has the [Location header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Location), if both things are true the response is a redirect.

**Reply by punkpeye** [reply]
How do I do this inside of `root.tsx` `loader`? attempting to access `response` headers is throwing an error.
